packages/@aws-cdk/aws-servicecatalogappregistry/README.md

```diff
@@ -64,14 +65,79 @@ const importedApplication = appreg.Application.fromApplicationArn(
 );
 ```
 
+## Automatic-Application
+
+An AppRegistry L2 construct to automatically create an application with the given name and description.
+The application name must be unique at the account level and it's immutable.
+`AutomaticApplication` L2 construct will automatically associate all stacks in the given scope, however
+in case of a `Pipeline` stack, stage underneath the pipeline will not automatically be associated and
+needs to be associated separately.
+If cross account stack is detected, then this construct will automatically share the application to consumer accounts.
+Cross account feature will only work for non environment agnostic stacks.
+
+Following will create an Application named `MyAutoApplication` in account `123456789012` and region `us-east-1`
+and will associate all stacks in the `App` scope to `MyAutoApplication`.
+
+```ts
+const app = new App();
+const autoApp = new appreg.AutomaticApplication(app, 'AutoApplication', {
+    applicationName: 'MyAutoApplication',
+    description: 'Testing auto application',
+    stackProps: {
+        stackName: 'MyAutoApplicationStack',
+        env: {account: '123456789012', region: 'us-east-1'},
+    },
+});
+```
+
+In case of a Pipeline stack, you need to pass the reference of `AutomaticApplication` to pipeline stack and associate
+each stage as shown below:
+
+```ts
+import * as cdk from "@aws-cdk/core";
+import * as codepipeline from "@aws-cdk/pipelines";
+import * as codecommit from "@aws-cdk/aws-codecommit";
+declare const repo: codecommit.Repository;
+declare const pipeline: codepipeline.CodePipeline;
+declare const beta: cdk.Stage;
+class ApplicationPipelineStack extends cdk.Stack {
+  constructor(scope: cdk.App, id: string, props: ApplicationPipelineStackProps) {
+    super(scope, id, props);
+    
+   //associate the stage to automatic application.
+   props.application.associateStage(beta, this.stackName);
+   pipeline.addStage(beta);
+  }
+};
+
+interface ApplicationPipelineStackProps extends cdk.StackProps {
+  application: appreg.AutomaticApplication;
+};
+
+const app = new App();
+const autoApp = new appreg.AutomaticApplication(app, 'AutoApplication', {
```

Comment:
Per what I see in the code, AutomaticApplication is a Construct. Construct's scope should be a Stack, not App. I think it makes more sense to have AppRegistry application to be part of the component's toolchain. The toolchain contains anything related to software development life cycle of the component. Some examples: 1/ pull request validation in trunk-based development model 2/ tenant provisioning logical unit in multi-tenant SaaS applications using silo deployment model. Here is an example of a toolchain containing continuous deployment and pull request validation logical units: https://github.com/alexpulver/usermanagement-backend/blob/main/toolchain.py

Reply:
An approach similar to [this](https://aws.amazon.com/blogs/mt/cross-account-configuration-with-aws-appconfig/) AWS AppConfig blog post which is aligned with [AWS mutli-account recommendations](https://docs.aws.amazon.com/whitepapers/latest/organizing-your-aws-environment/organizing-your-aws-environment.html).

Reply:
> Construct's scope should be a Stack, not App. 

I don't agree with this necessarily. A construct's scope can be any other construct, including an App. Yes, CFN resources ultimately need to be created in the (transitive) scope of a Stack, but this thing can be an exception.

Having said that, I think you actually mean something else. I *think* you mean something along the lines of:

*In the case of a Pipeline deployment, the Application should be created in the scope of the Pipeline Stack/Toolchain stack, not at the top level.*

I tend to agree that's what most people would probably want. But wouldn't that just mean you *don't* use `AutomaticApplication`, but use a plain old `Application` instead?